src/cascadia/TerminalSettingsModel/Profile.cpp

```diff
@@ -244,17 +245,30 @@ void Profile::LayerJson(const Json::Value& json)
 }
 
 // Method Description:
+// - Either Returns this profile's background image path, if one is set, expanding
 // - Returns this profile's background image path, if one is set, expanding
 //   any environment variables in the path, if there are any.
+// - Or if "DesktopWallpaper" is set, then gets the path to the desktops wallpaper.
 // Return Value:
-// - This profile's expanded background image path / the empty string.
+// - This profile's expanded background image path / desktops's wallpaper path /the empty string.
 winrt::hstring Profile::ExpandedBackgroundImagePath() const
 {
     if (_BackgroundImagePath.empty())
     {
         return _BackgroundImagePath;
     }
-    return winrt::hstring{ wil::ExpandEnvironmentStringsW<std::wstring>(_BackgroundImagePath.c_str()) };
+    // checks if the user would like to copy their desktop wallpaper
+    // if so, replaces the path with the desktop wallpaper's path
+    else if (_BackgroundImagePath == to_hstring(DesktopWallpaperEnum))
+    {
+        WCHAR desktopWallpaper[MAX_PATH];
+        SystemParametersInfo(SPI_GETDESKWALLPAPER, MAX_PATH, desktopWallpaper, SPIF_UPDATEINIFILE);

```

Comment:
I believe this function returns 0 or false in case of an error - I'd feel much better if we check that its return value wasn't that before we use `desktopWallpaper`

Reply:
"If there is no desktop wallpaper, the returned string is empty."

https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-systemparametersinfoa

If SPI_GETDESKWALLPAPER returns an empty string, would you like it to return something else other than an empty string?

Reply:
"If the function fails, the return value is zero" - if SystemParametersInfo fails

https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-systemparametersinfoa

Would you like us to just return an empty string if it fails?

Reply:
Yep, that makes sense! Thanks for the great work :)

Reply:
Should be pushed. Thanks